packages/store/src/usersSlice/index.ts

```diff
@@ -59,89 +135,88 @@ export const usersSlice = createSlice({
       state.byId[action.payload.id] = action.payload;
       state.currentUser = action.payload.id;
     },
-    createdUser: (state, action) => {
-      // the new user gets a 0 as id, since this will be overwritten by the retrieved userlist anyway + there is no way to know the id before
-      state.byId[0] = action.payload;
-    },
-    removedUser: (state, action) => {
+    removedUser: (state, action: PayloadAction<string>) => {
       // eslint-disable-next-line @typescript-eslint/no-unused-vars
       const { [action.payload]: removedUser, ...byId } = state.byId;
       state.byId = byId;
       state.currentUser = state.currentUser === action.payload ? null : state.currentUser;
     },
-    updatedUser: (state, action) => {
+    updatedUser: (state, action: PayloadAction<Partial<User> & { id: string }>) => {
       state.byId[action.payload.id] = {
         ...state.byId[action.payload.id],
         ...action.payload
       };
     },
-    receivedPermissionSets: (state, action) => {
+    receivedPermissionSets: (state, action: PayloadAction<Record<string, PermissionSet>>) => {
       state.permissionSetsById = action.payload;
     },
-    receivedRoles: (state, action) => {
+    receivedRoles: (state, action: PayloadAction<Record<string, Role>>) => {
       state.rolesById = action.payload;
+      state.rolesInitialized = true;
     },
-    createdRole: (state, action) => {
+    createdRole: (state, action: PayloadAction<Role>) => {
       state.rolesById[action.payload.name] = {
         ...state.rolesById[action.payload.name],
         ...action.payload
       };
     },
-    removedRole: (state, action) => {
+    removedRole: (state, action: PayloadAction<string>) => {
       // eslint-disable-next-line @typescript-eslint/no-unused-vars
       const { [action.payload]: toBeRemoved, ...rolesById } = state.rolesById;
       state.rolesById = rolesById;
     },
-    setCustomColumns: (state, action) => {
+    setCustomColumns: (state, action: PayloadAction<CustomColumn[]>) => {
       state.customColumns = action.payload;
     },
-    setGlobalSettings: (state, action) => {
+    setGlobalSettings: (state, action: PayloadAction<GlobalSettings>) => {
       state.settingsInitialized = true;
       state.globalSettings = {
         ...state.globalSettings,
         ...action.payload
       };
     },
-    setUserSettings: (state, action) => {
+    setUserSettings: (state, action: PayloadAction<UserSettings>) => {
       state.userSettingsInitialized = true;
       state.userSettings = {
         ...state.userSettings,
         ...action.payload
       };
     },
-    setTooltipState: (state, action) => {
+    setTooltipState: (state, action: PayloadAction<{ id: string; readState: ReadState }>) => {
       const { id, readState = READ_STATES.read } = action.payload;
-      state.tooltips.byId[id].readState = readState;
+      if (state.tooltips.byId[id]) {
+        state.tooltips.byId[id].readState = readState;
+      }
```

Comment:
```suggestion
      state.tooltips.byId[id] = { ...state.tooltips.byId[id], readState };
```
as changed tips will either exist already or if they don't (only reason I could see would be improperly stored  settings) we want to store the state instead of ignoring the setting